modules/local/bamtobed_sort.nf

```diff
@@ -0,0 +1,39 @@
+process BAMTOBED_SORT {
+    tag "$meta.id"
+    label "process_high"
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-9d3a458f6420e5712103ae2af82c94d26d63f059:60b54b43045e8cf39ba307fd683c69d4c57240ce-0' :
+        'quay.io/biocontainers/mulled-v2-9d3a458f6420e5712103ae2af82c94d26d63f059:60b54b43045e8cf39ba307fd683c69d4c57240ce-0' }"
+
+    input:
+    tuple val(meta), path(bam)
+
+    output:
+    tuple val(meta), path("*.bed"), emit: sorted_bed
+    path "versions.yml"           , emit: versions
+
+    script:
+    def prefix = args.ext.prefix ?: "${meta.id}"
+    """
+    samtools view -@4 -u -F0x400 ${bam} | bamToBed | sort -k4 --parallel=$task.cpus -S 50G > ${prefix}_merged_sorted.bed
```

Comment:
Can the cpu and memory assignments be made dynamic? I am worried what happens if someone only gives less than 50G of memory. Or 4 cpus or less, for example.

Reply:
Using any more than 4 cores for the samtools task will be very wastefull so i'll add in this:
```
    script:
    def prefix = args.ext.prefix ?: "${meta.id}"
    def st_cores = "${task.cpus}" > 4 ? 4 : "${task.cpus}"
    """
    samtools view -@${st_cores} -u -F0x400 ${bam} | bamToBed | sort -k4 --parallel=${task.cpus} -S ${task.memory}M > ${prefix}_merged_sorted.bed
    """
```